pkg/resource/config.go

```diff
@@ -184,54 +207,92 @@ func LoadAWSConfigFromAPIKeys(
 	token,
 	region,
 	roleArn,
+	roleSessionName,
 	externalId string,
 ) (*aws.Config, error) {
-	configOptions := []func(*config.LoadOptions) error{
-		config.WithRegion(region),
-		config.WithCredentialsProvider(
-			credentials.NewStaticCredentialsProvider(
-				accessKeyID,
-				secretAccessKey,
-				token,
+	configOptions := []func(*config.LoadOptions) error{}
+
+	// use specified region if provided
+	if region != "" {
+		configOptions = append(
+			configOptions,
+			config.WithRegion(region),
+		)
+	}
+
+	// use static credentials if provided
+	if accessKeyID != "" && secretAccessKey != "" {
+		configOptions = append(
+			configOptions,
+			config.WithCredentialsProvider(
+				credentials.NewStaticCredentialsProvider(
+					accessKeyID,
+					secretAccessKey,
+					token,
+				),
 			),
-		),
+		)
 	}
 
-	awsConfig, err := config.LoadDefaultConfig(
+	// use role session name if provided
+	if roleSessionName != "" {
+		configOptions = append(
+			configOptions,
+			config.WithAssumeRoleCredentialOptions(
+				func(o *stscreds.AssumeRoleOptions) {
+					o.RoleSessionName = roleSessionName
+				}),
+		)
+	}
+
+	config, err := config.LoadDefaultConfig(
 		context.Background(),
 		configOptions...,
 	)
 	if err != nil {
 		return nil, fmt.Errorf("failed to load AWS config from static API keys: %w", err)
 	}
+	awsConfig := &config
 
 	// assume role if roleArn is provided
 	if roleArn != "" {
-		awsConfig, err = assumeRole(roleArn, externalId, awsConfig, configOptions)
+		awsConfig, err = AssumeRole(roleArn, roleSessionName, externalId, 0, *awsConfig, configOptions)
 		if err != nil {
 			return nil, fmt.Errorf("failed to assume role: %w", err)
 		}
-		return &awsConfig, err
+		return awsConfig, err
 	}
 
-	return &awsConfig, err
+	return awsConfig, err
 }
 
-// assumeRole returns an AWS config with temporary credentials
+// AssumeRole returns an AWS config with temporary credentials
 // from an assumed role.
-func assumeRole(
+func AssumeRole(
 	roleArn,
+	roleSessionName,
 	externalId string,
+	sessionDuration int32,
 	awsConfig aws.Config,
 	configOptions []func(*config.LoadOptions) error,
-) (aws.Config, error) {
+) (*aws.Config, error) {
```

Comment:
Updated to return a pointer, which is consistent with other functions in this package that also return pointers to an `aws.Config` instead of a copy.